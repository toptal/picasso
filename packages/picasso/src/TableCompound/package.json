{
  "name": "@toptal/picasso-table-compound",
  "version": "1.0.0",
  "main": "index.js",
  "dependencies": {
    "@toptal/picasso-table": "^1.0.0",
    "@toptal/picasso-table-cell": "^1.0.0",
    "@toptal/picasso-table-body": "^1.0.0",
    "@toptal/picasso-table-row": "^1.0.0",
    "@toptal/picasso-table-head": "^1.0.0",
    "@toptal/picasso-table-section-head": "^1.0.0",
    "@toptal/picasso-table-footer": "^1.0.0",
    "@toptal/picasso-table-expandable-row": "^1.0.0"
  }
}