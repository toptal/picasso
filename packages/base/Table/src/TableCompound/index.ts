import { Table } from '../Table'
import { TableCell } from '../TableCell'
import { TableBody } from '../TableBody'
import { TableRow } from '../TableRow'
import { TableHead } from '../TableHead'
import { TableSectionHead } from '../TableSectionHead'
import { TableFooter } from '../TableFooter'
import { TableExpandableRow } from '../TableExpandableRow'

type TableCompoundType = typeof Table & {
  Cell: typeof TableCell
  Body: typeof TableBody
  Row: typeof TableRow
  Head: typeof TableHead
  SectionHead: typeof TableSectionHead
  Footer: typeof TableFooter
  ExpandableRow: typeof TableExpandableRow
}

export const TableCompound: TableCompoundType = Object.assign(Table, {
  Cell: TableCell,
  Body: TableBody,
  Head: TableHead,
  SectionHead: TableSectionHead,
  Row: TableRow,
  ExpandableRow: TableExpandableRow,
  Footer: TableFooter,
})
