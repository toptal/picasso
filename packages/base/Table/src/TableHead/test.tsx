import React from 'react'
import type { RenderResult } from '@toptal/picasso-test-utils'
import { render } from '@toptal/picasso-test-utils'

import { TableCompound as Table } from '../TableCompound'

const renderTableHead = () => {
  return render(
    <Table>
      <Table.Head>
        <Table.Row>
          <Table.Cell>Head test</Table.Cell>
        </Table.Row>
      </Table.Head>
    </Table>
  )
}

describe('TableHead', () => {
  let api: RenderResult

  beforeEach(() => {
    api = renderTableHead()
  })

  it('renders', () => {
    const { container } = api

    expect(container).toMatchSnapshot()
  })
})
