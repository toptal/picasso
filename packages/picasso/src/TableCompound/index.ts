import Table from '../Table'
import TableCell from '../TableCell'
import TableBody from '../TableBody'
import TableRow from '../TableRow'
import TableHead from '../TableHead'
import TableSectionHead from '../TableSectionHead'
import TableFooter from '../TableFooter'
import TableExpandableRow from '../TableExpandableRow'

export const TableCompound = Object.assign(Table, {
  Cell: TableCell,
  Body: TableBody,
  Head: TableHead,
  SectionHead: TableSectionHead,
  Row: TableRow,
  ExpandableRow: TableExpandableRow,
  Footer: TableFooter
})
