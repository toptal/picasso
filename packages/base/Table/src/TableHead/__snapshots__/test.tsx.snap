// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead renders 1`] = `
<div>
  <div
    class="Picasso-root"
  >
    <table
      class="w-full border-collapse border-spacing"
    >
      <thead
        class="border-0 border-solid border-b border-gray"
      >
        <tr
          class="border-0 border-solid border-b border-gray"
        >
          <th
            class="h-10 py-2 border-none last:pr-6 px-4 text-left"
          >
            <div
              class="m-0 text-xxs text-graphite font-semibold text-align"
            >
              Head test
            </div>
          </th>
        </tr>
      </thead>
    </table>
  </div>
</div>
`;
